src/object/obj_rpc.c

```diff
@@ -564,84 +506,50 @@ crt_proc_struct_obj_iod_array(crt_proc_t proc, struct obj_iod_array *iod_array)
 }
 
 static int
-crt_proc_daos_anchor_t(crt_proc_t proc, daos_anchor_t *anchor)
-{
-	if (crt_proc_uint16_t(proc, &anchor->da_type) != 0)
-		return -DER_HG;
-
-	if (crt_proc_uint16_t(proc, &anchor->da_shard) != 0)
-		return -DER_HG;
-
-	if (crt_proc_uint32_t(proc, &anchor->da_flags) != 0)
-		return -DER_HG;
-
-	if (crt_proc_raw(proc, anchor->da_buf, sizeof(anchor->da_buf)) != 0)
-		return -DER_HG;
-
-	return 0;
-}
-
-static int
-crt_proc_d_sg_list_t(crt_proc_t proc, d_sg_list_t *sgl)
+crt_proc_d_sg_list_t(crt_proc_t proc, d_sg_list_t *p)
 {
 	crt_proc_op_t	proc_op;
 	int		i;
 	int		rc;
 
-	rc = crt_proc_uint32_t(proc, &sgl->sg_nr);
+	rc = crt_proc_uint32_t(proc, &p->sg_nr);
 	if (rc != 0)
 		return -DER_HG;
 
-	rc = crt_proc_uint32_t(proc, &sgl->sg_nr_out);
+	rc = crt_proc_uint32_t(proc, &p->sg_nr_out);
 	if (rc != 0)
 		return -DER_HG;
 
 	rc = crt_proc_get_op(proc, &proc_op);
 	if (rc != 0)
 		return -DER_HG;
 
-	if (proc_op == CRT_PROC_DECODE && sgl->sg_nr > 0) {
-		D_ALLOC_ARRAY(sgl->sg_iovs, sgl->sg_nr);
-		if (sgl->sg_iovs == NULL)
+	if (proc_op == CRT_PROC_DECODE && p->sg_nr > 0) {
+		D_ALLOC_ARRAY(p->sg_iovs, p->sg_nr);
+		if (p->sg_iovs == NULL)
 			return -DER_NOMEM;
 	}
 
-	for (i = 0; i < sgl->sg_nr; i++) {
-		rc = crt_proc_d_iov_t(proc, &sgl->sg_iovs[i]);
+	for (i = 0; i < p->sg_nr; i++) {
+		rc = crt_proc_d_iov_t(proc, &p->sg_iovs[i]);
 		if (rc != 0) {
 			if (proc_op == CRT_PROC_DECODE)
```

Comment:
Not your actual change but seems strange we dont use macros already defined in obj_rpc.h such as DECODING, ENCODING, FREEING

Reply:
If I will need to update I can change this.